app/code/Magento/SearchStorefront/DataProvider/Product/LayeredNavigation/Builder/Category.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\SearchStorefront\DataProvider\Product\LayeredNavigation\Builder;
+
+use Magento\Framework\Api\Search\AggregationInterface;
+use Magento\Framework\Api\Search\AggregationValueInterface;
+use Magento\Framework\Api\Search\BucketInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\SearchStorefront\DataProvider\Category\Query\CategoryAttributeQuery;
+use Magento\SearchStorefront\DataProvider\CategoryAttributesMapper;
+use Magento\SearchStorefront\DataProvider\Product\LayeredNavigation\Formatter\LayerFormatter;
+use Magento\SearchStorefront\DataProvider\Product\LayeredNavigation\LayerBuilderInterface;
+use Magento\SearchStorefrontStore\Model\Store;
```

Comment:
it's better to use interface instead of particular implementation, moreover if you already have the interface.

Reply:
@lykhachov, if we don't need introduced interfaces and don't use them in codebase, please remove them at all.

Reply:
replaced with Interface